### PR TITLE
📖 Add Official Snekmate Style Guide

### LINKED ISSUE #36: Official Style Guide
🐍 snekmate does not currently have an official style guide for contributors to follow. Once we've implemented the `ERC-1155` and `ERC-4626` contracts and done a little cleanup, we should focus on the official style guide that needs to be followed. Importantly, it should be easily verifiable and enforceable via GitHub actions. There is an unofficial Vyper formatter called [mamushi](https://github.com/benber86/mamushi) (fork of the popular [Black](https://github.com/psf/black) formatter) that we should also investigate.

### PR DESCRIPTION
**Pull request by pcaversaccio**
Fixes #36.

### CODE REVIEW

File: GUIDELINES.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+The code should be simple and straightforward, with a focus on readability and comprehensibility. Consistency and predictability should be maintained throughout the code base. This is especially true for naming, which should be systematic, clear, and concise.
+
+Wherever possible, modularity, composability, and gas efficiency should be pursued, but not at the expense of security compromises.
+
+## 🛠 Pull Request (PR) Format
+
+Pull requests are _squash-merged_ in most cases to keep the `main` branch history clean. The title of the PR becomes the commit message, so it should be written in a consistent format:
+
+- Start with an emoji that well describes the PR.
```
Comment (jtriley-eth) [inline GUIDELINES.md:33]: Consider adding a specific list of items here. For example:

- 📝 - documentation
- ♻️ - refactor
- 👷 - ci/cd
- 🐛 - bug
- ✨ - feature add
- ⚡️ - optimization

File: GUIDELINES.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+- Do not end with a full stop (i.e. period).
+- Write in the imperative: "✨ Add Feature X" and not "✨ Adds Feature X" or "✨ Added Feature X".
+
+This repository does not follow conventional commits, so do not prefix the title with "fix:", "feat:", or similar. Also, pull requests in progress should be submitted as _Drafts_ and should not be prefixed with "WIP:" or similar.
+
+Branch names do not matter, and commit messages within a PR are mostly not important either, although they can support the review process.
+
+## 🐍Vyper Conventions
+
+- The names of `constant`, `immutable`, and state variables, functions, and function parameters use the snake case notation (e.g. `my_function`) if no other notation is enforced via an EIP standard.
```
Comment (jtriley-eth) [inline GUIDELINES.md:44]: constants and immutable values should be in "screaming snake case"

**Review by jtriley-eth — COMMENTED** [review]
Two recommended changes, though everything else looks good 👍

File: GUIDELINES.md (hunk 2 above)
Comment (pcaversaccio) [reply GUIDELINES.md]: noted and enhanced

File: GUIDELINES.md (hunk 1 above)
Comment (pcaversaccio) [reply GUIDELINES.md]: noted and enhanced. That's the list:
  - Bug Fix 🐛
  - CI/CD 👷‍♂️
  - Documentation 📖
  - Events 🔊
  - Gas Optimisation ⚡️
  - New Feature 💥
  - Nit 🥢
  - Refactor/Cleanup ♻️
  - Security Fix 🔒

**Pull request merged**